packages/cdktf/lib/tokens/private/resolve.ts

```diff
@@ -61,11 +62,16 @@ export function resolve(obj: any, options: IResolveOptions): any {
     const context: IResolveContext = {
       preparing: options.preparing,
       scope: options.scope,
+      suppressBraces: options.previousContext?.suppressBraces,
```

Comment:
Is this reset somewhere in the context chain if you have two nested contexts? E.g.: for `Fn.tolist(bar.baz, \`var-${index}\`)` should resolve to `"${tolist(element(foo.bar.baz, "val-${locals.index}"))}"` Sorry, can't escape the backticks in the first example

Reply:
It's not reset, but `TFExpression.resolveString()` has code to handle adding them based on the tokens that exist.

Reply:
Ah nice 👍